src/Features/VisualBasic/Portable/CodeFixes/AddImport/VisualBasicAddImportCodeFixProvider.vb

```diff
@@ -297,17 +314,56 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeFixes.AddImport
                 placeSystemNamespaceFirst As Boolean,
                 cancellationToken As CancellationToken) As Task(Of Document)
 
+            Dim nameSyntax = DirectCast(symbol.GenerateTypeSyntax(addGlobal:=False), NameSyntax)
+
+            Return Await AddImportsAsync(
+                contextNode, document, placeSystemNamespaceFirst, nameSyntax,
+                additionalAnnotation:=Nothing, cancellationToken:=cancellationToken).ConfigureAwait(False)
+        End Function
+
+        Private Shared Async Function AddImportsAsync(
+                contextNode As SyntaxNode,
+                document As Document,
+                placeSystemNamespaceFirst As Boolean,
+                nameSyntax As NameSyntax,
+                additionalAnnotation As SyntaxAnnotation,
+                cancellationToken As CancellationToken) As Task(Of Document)
             Dim root = DirectCast(Await contextNode.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(False), CompilationUnitSyntax)
 
             Dim memberImportsClause =
-                SyntaxFactory.SimpleImportsClause(name:=DirectCast(symbol.GenerateTypeSyntax(addGlobal:=False), NameSyntax).WithAdditionalAnnotations(Simplifier.Annotation))
+                SyntaxFactory.SimpleImportsClause(name:=nameSyntax.WithAdditionalAnnotations(Simplifier.Annotation))
             Dim newImport = SyntaxFactory.ImportsStatement(
                 importsClauses:=SyntaxFactory.SingletonSeparatedList(Of ImportsClauseSyntax)(memberImportsClause))
 
+            If additionalAnnotation IsNot Nothing Then
+                newImport = newImport.WithAdditionalAnnotations(additionalAnnotation)
+            End If
+
+            ' Don't add the import if an eqiuvalent one is already there.
+            If root.Imports.Any(Function(i) i.IsEquivalentTo(newImport, topLevel:=False)) Then
```

Comment:
Yes.  IsEquivalent purely checks syntactic structure.  It doesn't check annotatoins.